File: src/main/java/cz/stasimek/fakturaceeasypeasy/entity/interfaces/Identifiable.java
package cz.stasimek.fakturaceeasypeasy.entity.interfaces;

import java.io.Serializable;

/**
 * An interface for objects with an {@code id} property.
 *
 * @param <I> the type of the entity id
 */
public interface Identifiable<I extends Serializable> {

	I getId();

	void setId(I id);
}

File: src/main/kotlin/cz/stasimek/fakturaceeasypeasy/entity/interfaces/Identifiable.kt
package cz.stasimek.fakturaceeasypeasy.entity.interfaces

import java.io.Serializable
import java.util.*

/**
 * An interface for objects with an `id` property.
 *
 * @param <I> the type of the entity id
</I> */
interface Identifiable<I : Serializable?> {
    var id: I
}